pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/MergeRollupMinionClusterIntegrationTest.java

```diff
@@ -0,0 +1,565 @@
+    //    {myTable2_16071_16101_3_1, myTable2_16071_16101_3_2, myTable2_16102_16129_4_1, myTable2_16102_16129_4_2,
+    //     myTable2_16130_16159_5_1, myTable2_16130_16159_5_2, myTable2_16160_16189_6_1, myTable2_16160_16189_6_2
+    //     myTable2_16190_16220_7}
+    //      -> {merged_150days_T1_0_myTable2_16065_16198_0, merged_150days_T1_0_myTable2_16205_16219_1}
+    // 2.
+    //    {merged_150days_T1_0_myTable2_16205_16219_1, myTable2_16221_16250_8_1, myTable2_16221_16250_8_2,
+    //     myTable2_16251_16281_9_1, myTable2_16251_16281_9_2, myTable2_16282_16312_10_1
+    //     myTable2_16282_16312_10_2, myTable2_16313_16342_11_1, myTable2_16313_16342_11_2,
+    //     myTable2_16343_16373_0_1, myTable2_16343_16373_0_2}
+    //      -> {merged_150days_1628644088146_0_myTable2_16205_16345_0,
+    //          merged_150days_1628644088146_0_myTable2_16352_16373_1}
+    // 3.
+    //    {merged_150days_1628644088146_0_myTable2_16352_16373_1, myTable2_16374_16404_1_1, myTable2_16374_16404_1_2
+    //     myTable2_16405_16435_2_1, myTable2_16405_16435_2_2}
+    //      -> {merged_150days_1628644105127_0_myTable2_16352_16429_0}
+
+    String sqlQuery = "SELECT count(*) FROM myTable2 LIMIT 200000"; // 115545 rows for the test table
+    JsonNode expectedJson = postSqlQuery(sqlQuery, _brokerBaseApiUrl);
+    int[] expectedNumSegmentsQueried = {16, 7, 3};
+    int[] expectedNumMergedSegments = {2, 4, 5};
+    long expectedWatermark = 16050 * 86_400_000L;
+    String offlineTableName = TableNameBuilder.OFFLINE.tableNameWithType(SINGLE_LEVEL_ROLLUP_TEST_TABLE);
+    int numTasks = 0;
+    for (String tasks = _taskManager.scheduleTasks(Collections.singletonList(offlineTableName), true)
+        .get(MinionConstants.MergeRollupTask.TASK_TYPE); tasks != null; tasks =
+        _taskManager.scheduleTasks(Collections.singletonList(offlineTableName), true)
+            .get(MinionConstants.MergeRollupTask.TASK_TYPE), numTasks++) {
+      Assert.assertEquals(_helixTaskResourceManager.getTaskConfigs(tasks).size(), 1);
+      Assert.assertTrue(_helixTaskResourceManager.getTaskQueues()
+          .contains(PinotHelixTaskResourceManager.getHelixJobQueueName(MinionConstants.MergeRollupTask.TASK_TYPE)));
+      // Will not schedule task if there's incomplete task
+      Assert.assertNull(_taskManager.scheduleTasks(Collections.singletonList(offlineTableName), true)
+          .get(MinionConstants.RealtimeToOfflineSegmentsTask.TASK_TYPE));
+      waitForTaskToComplete();
+
+      // Check watermark
+      MergeRollupTaskMetadata minionTaskMetadata = MergeRollupTaskMetadata.fromZNRecord(
+          _taskManager.getClusterInfoAccessor().getMinionMergeRollupTaskZNRecord(offlineTableName));
+      assertNotNull(minionTaskMetadata);
+      Assert.assertEquals((long) minionTaskMetadata.getWatermarkMap().get("150days"), expectedWatermark);
+      expectedWatermark += 150 * 86_400_000L;
+
+      // Check num total doc of merged segments are the same as original segment
```

Comment:
Why the number of rows for the roll-up test is the same?

Roll-up can aggregate the rows so the result may be smaller. I think that we should test the case where we reduce the number of rows after rollup. I thought we agreed to push the same data x2 and run the roll-up.

Reply:
The comment is wrong, updated.